### PR TITLE
Run docs and man generation in CI

### PR DESCRIPTION
**Pull request by dnephin**
This way if we make changes to the scripts, or a docs change is invalid somehow we might be able to catch that problem in CI. We could probably do a little more validation on the output, but this is better than nothing.

Also cleanup the scripts a bit to be more consistent, and fail on errors.

### CODE REVIEW

File: scripts/docs/generate-man.sh [hunk 1]
```diff
@@ -1,14 +1,11 @@
...
-
-set -eu
+# Generate man pages for docker/cli
+set -eu -o pipefail
 
 mkdir -p ./man/man1
 
 MD2MAN_REPO=github.com/cpuguy83/go-md2man
-MD2MAN_COMMIT=$(grep -F "$MD2MAN_REPO" vendor.conf | cut -d' ' -f2)
+MD2MAN_COMMIT=v1.0.6
```
Comment (thaJeztah) [inline scripts/docs/generate-man.sh]: Should we bump the version in vendor.conf to the same? Was there a specific reason to not keep them in sync?

Comment (dnephin) [inline scripts/docs/generate-man.sh]: There is no version in vendor.conf anymore. If there was we could remove this `go get` entirely.

Comment (dnephin) [inline scripts/docs/generate-man.sh]: Ok, apparently it is in the `vendor.conf` ... let me try this again.

**Comment by dnephin** [comment]
@thaJeztah thanks! I've fixed this up to use the code we already have in vendor. This is also faster because we don't have to go downloading anything.

**Pull request merged**